seqlog/structured_logging.py

```diff
@@ -427,3 +435,44 @@ def _get_local_timestamp(record):
     )
 
     return timestamp.isoformat(sep=' ')
+
+
+def _ensure_class(class_or_class_name, compatible_class=None):
+    """
+    Ensure that the supplied value is either a class or a fully-qualified class name.
+
+    :param name: A class or a fully-qualified class name.
+    :param expected_base_class: If specified then the class must be, or (eventually) derive from, this class.
+    :return: The class represented by class_or_class_name.
+    """
+
+    target_class = class_or_class_name
+    if isinstance(class_or_class_name, str):
+        name_parts = class_or_class_name.split('.')
+        module_name = '.'.join(
+            name_parts[:-1]
+        )
+        target_class_name = name_parts[-1:]
+
+        # Raises ModuleNotFoundError if we can't resolve part of the module path
+        target_module = importlib.import_module(module_name)
+
+        target_class = getattr(target_module, target_class_name, default=None)
```

Comment:
raise the error :
`TypeError: getattr() takes no keyword arguments`
my python version is 3.70
the problem is solved when i modify the code to:
`target_class = getattr(target_module, target_class_name)`

Reply:
if you want to use the build-in class `BaseConfigurator`, we could add some code to confirm the custom json encoder is a sub class of `json.encoder.JSONEncoder`:
```python3
def configure_from_file(file_name, override_root_logger=True, use_structured_logger=True):
    """your comment"""
       with open(file_name) as config_file:
        config = yaml.load(config_file)
    for name, handler in config.get('handlers', {}).items():
        if 'json_encoder_class' in handler:
            json_encoder_config = logging.config.BaseConfigurator({})
            custom_encoder = json_encoder_config.resolve(handler['json_encoder_class'])
            if not issubclass(custom_encoder, json.encoder.JSONEncoder):
                raise TypeError("Invalid json encoder class {}, please use the class based on json.encoder.JSONEncoder!".format(custom_encoder))
            config['handlers'][name]['json_encoder_class'] = custom_encoder

    configure_from_dict(config, override_root_logger)
```